src/cli/src/cluster/install/k8.rs

```diff
@@ -1,13 +1,12 @@
-use std::io::Error as IoError;
-use std::io::ErrorKind;
 use std::convert::TryInto;
 use std::process::Command;
+use anyhow::Context;
 
 use fluvio_cluster::ClusterInstaller;
 use fluvio::config::TlsPolicy;
 use super::*;
 
-pub async fn install_core(opt: InstallCommand) -> Result<(), CliError> {
```

Comment:
is there reason for removing type signature?

Reply:
So here, `CliError` was doing nothing but catching the custom `IoError` from the git_hash line. `eyre/anyhow` can catch any error type that impl's std::error::Error, so this is pretty much equivalent. In a different case, it may be useful to manually create a `CliError` instance before returning it, but there was no extra value that would have added here.

Reply:
Reason for this is to statistically determine error during compile time as noted by rust-lang doc: https://doc.rust-lang.org/stable/rust-by-example/error/multiple_error_types/boxing_errors.html.   Yes it is more verbose and work but we want to preserve Rust's compile type checking much as possible

Reply:
I agree with you in the context of library errors, where the structure of the errors and the internal source errors matters. That's why I didn't use anyhow/eyre in the library crates, only in the CLI crate. Note that in any of these functions returning an `eyre::Result`, we _could_ construct a `CliError` and return it, using `?`'s From-coercion to convert the type. This would be the case if we were to do something like this:

```rust
fn some_function(arg: i32) -> eyre::Result<()> {
    if arg < 0 {
        return Err(CliError::invalid_arg("Arg cannot be less than 0!").into());
    }
    Ok(())
}
```

Then the structure of the error is still preserved, i.e. the Display of the `CliError::InvalidArg` is shown in the error stack.

But overall, libraries mostly _produce_ errors, so they should be well-structured and typed, which is what `thiserror` is good at helping with. Applications mostly _consume_ errors and are interested in reporting them, which is what the `anyhow/eyre` crates are good at, and why it's fine to use `eyre::Result<T>` to capture any error. https://nick.groenen.me/posts/rust-error-handling/#libraries-versus-applications

Reply:
Separation between library and application is not so clear since some of CLI could be turn into library as happened with Cluster crate.   I don't see clear justification for changing this except to save couple of lines and don't know long term consequence of losing concrete error type.  So Let's keep status quo until it is more clear.